pkg/util/template/resource.go

```diff
@@ -67,41 +94,73 @@ type PlainText struct {
 	ComponentDependencies []*PlainText
 }
 
+type MessagePlainText struct {
+	// Name is the name of template
+	Name string
+}
+
 var _ resource.Descriptor = &PlainText{}
+var _ resource.Descriptor = &MessagePlainText{}
 
-func (t *PlainText) templateResource() {}
+func (t *PlainText) templateResource()        {}
+func (t *MessagePlainText) templateResource() {}
 
 func (t *PlainText) MatchResource(path string) (*resource.Match, bool) {
+	if strings.HasPrefix(t.Name, "messages/") {
+		return nil, false
+	}
+	return matchTemplatePath(path, t.Name)
+}
+
+func (t *MessagePlainText) MatchResource(path string) (*resource.Match, bool) {
+	if !strings.HasPrefix(t.Name, "messages/") {
+		return nil, false
+	}
 	return matchTemplatePath(path, t.Name)
 }
 
 func (t *PlainText) FindResources(fs resource.Fs) ([]resource.Location, error) {
+	// Exclude App Fs
+	if fs.GetFsLevel() == resource.FsLevelApp {
+		return []resource.Location{}, nil
+	}
+	return readTemplates(fs, t.Name)
+}
+
+func (t *MessagePlainText) FindResources(fs resource.Fs) ([]resource.Location, error) {
+	// Any Fs
 	return readTemplates(fs, t.Name)
 }
 
 func (t *PlainText) ViewResources(resources []resource.ResourceFile, view resource.View) (interface{}, error) {
 	return viewTextTemplates(t.Name, resources, view)
 }
 
+func (t *MessagePlainText) ViewResources(resources []resource.ResourceFile, view resource.View) (interface{}, error) {
+	return viewTextTemplates(t.Name, resources, view)
+}
+
 func (t *PlainText) UpdateResource(_ context.Context, _ []resource.ResourceFile, resrc *resource.ResourceFile, data []byte) (*resource.ResourceFile, error) {
+	return nil, fmt.Errorf("PlainText resource cannot be updated. Use MessagePlainText resource instead.")
```

Comment:
Allow PlainText to be updated, same as before.

Reply:
reverted